bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/datamodel/FreeAtHomeDatapointGroup.java

```diff
@@ -0,0 +1,288 @@
+        return PidTranslationUtils.getDescriptionTextForPairingId(String.format("0x%04X", pairingId));
+    }
+
+    public String getValueType() {
+        return PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+    }
+
+    public String getTypePattern() {
+        String pattern = "";
+
+        functionString = PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+
+        switch (functionString) {
+            case PidTranslationUtils.PID_VALUETYPE_DECIMAL:
+                pattern = "%.1f";
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_INTEGER:
+                pattern = "%d";
+                break;
+            default:
+                pattern = "";
+                break;
+        }
+
+        return pattern;
+    }
+
+    @SuppressWarnings("null")
+    public ValueStateConverter getValueStateConverter() {
+        ValueStateConverter valueStateConverter = new BooleanValueStateConverter();
+
+        functionString = PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+
+        switch (functionString) {
+            case PidTranslationUtils.PID_VALUETYPE_BOOLEAN:
+                valueStateConverter = new BooleanValueStateConverter();
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_DECIMAL:
+                valueStateConverter = new DecimalValueStateConverter();
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_INTEGER:
+                valueStateConverter = new DecimalValueStateConverter();
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_STRING:
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_SHUTTERMOVEMENT:
+                valueStateConverter = new ShuttercontrolValueStateConverter();
+                break;
+            default:
+                valueStateConverter = new BooleanValueStateConverter();
+                break;
+        }
+
+        return valueStateConverter;
+    }
+
+    public String getOpenHabItemType() {
+        String itemTypeString = "";
+
+        functionString = PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+
+        switch (functionString) {
+            case PidTranslationUtils.PID_VALUETYPE_BOOLEAN:
+                itemTypeString = CoreItemFactory.SWITCH;
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_DECIMAL:
+                itemTypeString = CoreItemFactory.NUMBER;
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_INTEGER:
+                itemTypeString = CoreItemFactory.NUMBER;
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_SHUTTERMOVEMENT:
+                itemTypeString = CoreItemFactory.ROLLERSHUTTER;
+                break;
+            default:
+                logger.info("wrong datapoint {}", String.format("0x%04X", pairingId));
```

Comment:
What will happen in this case? Maybe it would be worth throwing some sort a `RuntimeException` (e.g. `IllegalStateException`) if this is not expected to happen?